### ISSUE TITLE
[Android] No signature of method

### ISSUE THREAD

**Issue by oleksiiSevrukovRedvike** [twilio/twilio-voice-react-native#380]
# Issue

## Pre-submission Checklist
- [x] I have verified that the issue occurs with the latest release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice-react-native/blob/main/CHANGELOG.md).
- [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice-react-native/blob/main/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
- [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

## Description 
I've added twilio-voice-react-native and I can't sync android project now. I get error
```
A problem occurred evaluating project ':twilio_voice-react-native-sdk'.
> No signature of method: build_bawhh4j7dlbd0hfyx0jm5xngf.android() is applicable for argument types: (build_bawhh4j7dlbd0hfyx0jm5xngf$_run_closure1) values: [build_bawhh4j7dlbd0hfyx0jm5xngf$_run_closure1@230732e0]
```

## Reproduction Steps
1. Install twilio-voice-react-native
2. Press "Sync Project with Gradle Files"
3. See error

## Expected Behavior
Android project can be synced

## Actual Behavior
I get error on sync step

## Reproduction Frequency
100%

## Software and Device Information
**Please complete the following information.**
 - React version: [17.0.1]
 - React Native version: [0.64.0]
 - Node version: [16.20.2]
 - yarn version: [1.22.19]
 - twilio [3.72.0]
 - @twilio/conversations [2.1.0]
 - @twilio/voice-react-native-sdk [1.0.0]
 - @react-native-firebase/app [20.1.0]
 - @react-native-firebase/firestore [20.1.0]
 - @react-native-firebase/messaging [20.1.0]
 - java [15.0.10] (azul-15)
 - com.android.tools.build:gradle [4.1.0]
 - com.google.gms:google-services [4.3.15]
 - minSdkVersion [21]
 - compileSdkVersion [33]
 - targetSdkVersion [33]
 - ndkVersion [23.1.7779620]
 - kotlinVersion [1.8.10]
 - buildToolsVersion [33.0.0]

**Comment by mhuynh5757** [comment]
Hi @oleksiiSevrukovRedvike thanks for reaching out. The version details and issue description are much appreciated, thank you. Just to confirm, you were able to build and run your app before installing the Twilio RN SDK as a dependency? And you only started to see this sync issue _after_ you installed the library? Could you try doing a gradle clean using Android Studio, and attempt to build again? If that fails, could you try to remove your `node_modules` folder, do a clean `yarn install`, and try building again?

**Comment by oleksiiSevrukovRedvike** [comment]
Hi @mhuynh5757!
Yes, previously sync and build were successful without Twilio RN SDK.
Yes, this error appears only after installation of Twilio RN SDK.
Gradle clean didn't helped.
Remove node_modules folder and clean yarn install didn't helped too.

**Comment by oleksiiSevrukovRedvike** [comment]
@mhuynh5757 if I remove `namespace 'com.twiliovoicereactnative'` from build.gradle sync is successful
But when I try to make build I get error `error: package android.app does not exist` in `VoiceApplicationProxy.java:8` so it can't import `android.app.Application;`

**Comment by oleksiiSevrukovRedvike** [comment]
@mhuynh5757 if I change `JavaVersion.VERSION_11` to `JavaVersion.VERSION_1_8` build is successful but I get crash immediately with error `java.lang.NoClassDefFoundError: Failed resolution of: Lkotlin/collections/CollectionsKt;`
I have downgraded firebase libraries to versions where they use `JavaVersion.VERSION_1_8`, but do you have releases with `JavaVersion.VERSION_1_8` ?

**Comment by mhuynh5757** [comment]
Hi @oleksiiSevrukovRedvike thanks for providing more info. Apologies for the delayed response, we'll take a look at this issue as soon as we can.

**Comment by oleksiiSevrukovRedvike** [comment]
@mhuynh5757 I've solved this issue by upgrading react native in our project to 72 and now it's building. So probably you need to update docs with minimal react native version support, which support JavaVersion.VERSION_11

**Comment by mhuynh5757** [comment]
Internal ticket filed for this issue: https://twilio-engineering.atlassian.net/browse/VBLOCKS-4176
